db/config.cc

```diff
@@ -842,6 +842,10 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , max_memory_for_unlimited_query_hard_limit(this, "max_memory_for_unlimited_query_hard_limit", "max_memory_for_unlimited_query", liveness::LiveUpdate, value_status::Used, (uint64_t(100) << 20),
             "Maximum amount of memory a query, whose memory consumption is not naturally limited, is allowed to consume, e.g. non-paged and reverse queries. "
             "This is the hard limit, queries violating this limit will be aborted.")
+    , reader_concurrency_semaphore_soft_limit_multiplier(this, "reader_concurrency_semaphore_soft_limit_multiplier", liveness::LiveUpdate, value_status::Used, 1,
+            "Start serializing reads after their collective memory consumption goes above $normal_limit * $multiplier.")
+    , reader_concurrency_semaphore_hard_limit_multiplier(this, "reader_concurrency_semaphore_hard_limit_multiplier", liveness::LiveUpdate, value_status::Used, 3,
+            "Start killing reads after their collective memory consumption goes above $normal_limit * $multiplier.")
```

Comment:
Let's call it serializing_limit and kill_limit so we don't have to internally translate it.

Reply:
and, should it be a multiplier or just byte count?

Reply:
What do you mean by a byte count? A memory amount? We can't do that, we use some percent of shard's memory for semaphore's memory amount. Furthermore, on enterprise we have per-service level semaphores with share proportional memory accounts.

Reply:
I meant a memory amount (that has default values according to an initializer). But let's keep it as is.